datalake/src/main/java/com/sequenceiq/datalake/flow/detach/event/DatalakeResizeRecoveryFlowChainStartEvent.java

```diff
@@ -0,0 +1,40 @@
+package com.sequenceiq.datalake.flow.detach.event;
+
+import java.util.Objects;
+
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.flow.SdxEvent;
+
+public class DatalakeResizeRecoveryFlowChainStartEvent extends SdxEvent {
+    public static final String SDX_RESIZE_RECOVERY_FLOW_CHAIN_START_EVENT = "DatalakeResizeRecoveryFlowChainStartEvent";
+
+    private final SdxCluster oldCluster;
+
+    private final SdxCluster newCluster;
+
+    public DatalakeResizeRecoveryFlowChainStartEvent(SdxCluster oldCluster, SdxCluster newCluster, String userId) {
```

Comment:
I'm sure the recovery that triggers this flow chain by sending the event would have information about the old datalake.

It should be implementation details. All the recovery API would about is the datalake that they want to recover. Based on the state if the datalake appropriate recovery mechanism would be triggered.

Reply:
I don't understand what you mean.